### ISSUE TITLE
AFK move/kick not working

### ISSUE THREAD

**Issue by SupaFred** [Person8880/Shine#839]
I've configured AFK kick to move players to spectate after 50 seconds of inactivity but since a little while back it's not working. From the logs it seems like the plugin thinks they are already on spectate.

[18:04:03][AFKKick] [Debug] Applying normal warn actions (MOVE_TO_SPECTATE, NOTIFY) to Bob Saget[41422823] as their last move time was 73.50 vs. 131.79 (no movement for 58.30 seconds).
[18:04:03][AFKKick] [Debug] Applying normal warn actions (MOVE_TO_SPECTATE, NOTIFY) to |-DMD-| BLOCKY_DUDE[923742015] as their last move time was 23.35 vs. 131.79 (no movement for 108.44 seconds).
[18:04:03][AFKKick] [Debug] Applying normal warn actions (MOVE_TO_SPECTATE, NOTIFY) to hal-one[149864897] as their last move time was 25.20 vs. 131.79 (no movement for 106.59 seconds).
[18:04:05]Chat from u[53583][Marines]: lol outplayed hte chair
[18:04:12][Shuffle] [Debug] Permitting vote as average difference is: 240.44444444444
[18:04:12]Rundervinkje[1042213] ran command sh_voterandom with no arguments.
[18:04:13][AFKKick] [Debug] Skipping movement of Bob Saget[41422823] as they are already on team 3.
[18:04:13][AFKKick] [Debug] Skipping movement of hal-one[149864897] as they are already on team 3.
[18:04:13][AFKKick] [Debug] Skipping movement of |-DMD-| BLOCKY_DUDE[923742015] as they are already on team 3.

**Comment by Person8880** [comment]
I can't reproduce this, and that log message can only occur if the game considers the player on the spectator team. Are you sure they were not already on the spectator team? Does it always log this?

**Comment by SupaFred** [comment]
Yes I'm sure they were on a team and not spectating. It seems like the afk plugin rarely works for me since a couple of weeks back. I've checked the logs a few times and seen the same messages. Do you want a full log?

**Comment by Person8880** [comment]
A full log from when this problem occurs may be helpful.

**Comment by SupaFred** [comment]
I keep having problems with the AFK kick not always working as intended. Here's a log from two days ago: https://pastebin.com/TuWEt4EE
The round started at 20:08 and Syntax Of Limbo was AFK the entire time. At 20:12 I moved him manually to spectate and he was warned at 20:13 and kicked at 20:14. I've configured the plugin to warn after about a minute and move the player to spectate and then kick them after 2 minutes. It seems like the AFK plugin wasn't working for Syntax until after I moved him to spectate.

Here is my AFK plugin config https://pastebin.com/sQUg4VZ0

**Comment by Person8880** [comment]
I'm pretty sure I found the problem. The plugin was still checking for old flags which are no longer implemented, including one that used to be used to denote a player that was frozen during the game start countdown. The flag is still set by the game code when a player joins a team during the countdown, but it's no longer removed so they are considered frozen for the entire round.

I've stopped it checking for the old flags now and also improved its detection of the concede sequence which it also wasn't detecting. Hopefully that fixes the issue. I'll likely release the changes tomorrow at some point.

I also added a new command, `sh_afk_status`, which may be useful in the future if an issue like this happens again. It prints the current state of all tracked players, including how long the plugin thinks they've been AFK, when they last moved, and what their last action was.
